OMGVA PoS/Business layer/Controllers/BusinessController.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using OMGVA_PoS.Data_layer.DTOs;
+using OMGVA_PoS.Data_layer.Enums;
+using OMGVA_PoS.Data_layer.Models;
+using OMGVA_PoS.Data_layer.Repositories.Business_Management;
+using OMGVA_PoS.Helper_modules.Utilities;
+using System.IdentityModel.Tokens.Jwt;
+
+namespace OMGVA_PoS.Business_layer.Controllers
+{
+    [Route("business")]
+    [ApiController]
+    public class BusinessController(IBusinessRepository businessRepository) : Controller
+    {
+        private readonly IBusinessRepository _businessRepository = businessRepository;
+
+        [HttpGet]
+        [Authorize(Roles = "Admin")]
+        public IActionResult GetAllBusinesses() {
+            return Ok(JsonConvert.SerializeObject(_businessRepository.GetBusinesses()));
+        }
+
+
+        [HttpGet("{id}")]
+        [Authorize(Roles = "Admin,Owner")]
+        public IActionResult GetBusiness(long id) {
+            JwtSecurityToken token = JwtTokenHelper.GetJwtToken(HttpContext.Request.Headers.Authorization);
+            if (token.UserRoleEquals(UserRole.Owner) && !token.UserBusinessEquals(id)) {
+                return Forbid();
```

Comment:
For all return statuses, I recommend putting the "ProducesResponseType" class attribute above the method. This will ensure that possible errors will be seen on Swagger. In particular, to see that this method can end with a "Forbidden" error, you might want to add the [ProducesResponseType(typeof(string), StatusCodes.Status403Forbidden)] attribute above the method.